posthog-react-native/src/surveys/PostHogSurveyProvider.tsx

```diff
@@ -32,7 +32,11 @@ const FeedbackSurveyContext = React.createContext<
 export const useFeedbackSurvey = (selector: string): FeedbackSurveyHook | undefined => {
   const context = React.useContext(FeedbackSurveyContext)
   const survey = context?.surveys.find(
-    (survey: Survey) => survey.type === SurveyType.Widget && survey.appearance?.widgetSelector === selector
+    (survey: Survey) =>
+      survey.type === SurveyType.Widget &&
+      survey.appearance?.widgetSelector === selector &&
+      survey.start_date !== null &&
+      survey.end_date === null
```

Comment:
I think we should call `getActiveMatchingSurveys` instead, passing `context?.surveys`.
Since the survey should only be shown in certain conditions, and this is a subset of it but, everything should apply right, even if you call `useFeedbackSurvey` yourself.

Reply:
Yeah, I guess that makes sense... 🤔 

For context, my scenario for using this hook is that I have UI in our app where a customer can explicitly open a survey. e.g.

- A "Submit feedback" button in the menu which opens a generic feedback survey. If survey doesn't exist, fall back to emailing us
- A "Request another integration" button which appears at the end of a list of integrations. If survey doesn't exist, just don't show the button.

Because of the fallbacks I can't just use event triggers and a noop button, and I figure some devs may want a custom UI that just submits data back to posthog. I hadn't run into the need for all the conditions `getActiveMatchingSurveys` handles, but thinking about it now I don't think it'd hurt and may be useful for some people.